compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp

```diff
@@ -557,6 +584,117 @@ struct QuantizeWeights final : public luci::CircleNodeMutableVisitor<bool>
   }
 };
 
+/**
+ * @brief QuantizeConstInputs quantizes const input tensors
+ * @details Quantize values using min/max of const values
+ */
+struct QuantizeConstInputs final : public luci::CircleNodeMutableVisitor<bool>
+{
+  QuantizeConstInputs(loco::DataType input, loco::DataType output)
+      : input_type(input), output_type(output)
+  {
+  }
+
+  loco::DataType input_type;
+  loco::DataType output_type;
+
+  bool visit(luci::CircleNode *node)
+  {
+    LOGGER(l);
+    INFO(l) << "QuantizeConstInputs visit node: " << node->name() << std::endl;
+    auto opcode = node->opcode();
+    auto arity = node->arity();
+
+    loco::Node *input_node{nullptr};
+    luci::CircleConst *const_node{nullptr};
+
+    switch (opcode)
```

Comment:
It's quite strange to have a `switch` inside a visitor 😄  Either of them is superflous.

Reply:
I've changed the code not to use visitor pattern.

Reply:
Thanks. Personally I'd go with visitor (I don't like big functions and especially long switchs).

Reply:
I'd like to use a function here because implementing a visitor for each operator would make lots of duplicate codes.